### DISCUSSION TITLE
Using two or more fonts at the same codepoint

### DISCUSSION THREAD

**Discussion by brainwo** [fdw/rofimoji#153] [Ideas]
Recently I found an [icon set font](https://github.com/google/material-design-icons/blob/master/variablefont/MaterialSymbolsOutlined%5BFILL%2CGRAD%2Copsz%2Cwght%5D.codepoints) with huge codepoint range (E003-F8FF),
it overlaps with my [other font](https://www.nerdfonts.com/) I use to theme Rofi in my `theme.rasi`.

I don't find any other solution than having two themes and have two shortcuts to launch rofimoji.
I would like to have everything in one place, is there any possibility to do this? :thinking:

**Comment by fdw** [comment]
I don't think I understand you yet, sorry... the two fonts you linked have different icons at the same codepoints, and so you would like to use `rofimoji` to show both at the same time? Is that correct?

If so, I don't see a way currently, as `rofi` can only use one font, so you'd always have to choose. Or, you accept that for one of the fonts, an incorrect glyph is shown.

Additionally, I don't think there's a csv yet for the material design icon font; we'd have to create that first.

**Reply by brainwo** [reply]
Yes, exactly. Two fonts at the same codepoints.

I was thinking if there is no solution for this yet... there are two ways to achieve this:
1. Caching everything as SVG (or a raster format, whichever Rofi supports)
   Newly added fonts will first be converted to SVG then cached for future use.
   Though, I'm not really sure about this, it might be better if the glyph can be read on the fly and converted into some other format directly, just like creating thumbnails for file browsers.
1. If Rofi rasi supports classes like CSS does, it could be helpful if you can have it like:
```css
element-text.nerd {
  font: "Nerd Font 12"
}
element-text.material {
  font: "Material Font 12"
}
```

I have made the [CSV for Material Symbols](https://gist.github.com/brainwo/ebebd105e81e941cd2f91ed4ca7ef561) before.

**Reply by fdw** [reply]
> Caching everything as SVG (or a raster format, whichever Rofi supports)

While that could work, for example using rofi's icons, it feels like a lot of overhead and complexity to create these images using the correct font and maintain them. Additionally, the user still has to take care that the inserted character is shown using the correct font: Imagine you chose one from NerdFonts, but the receiving application is set to Material Icons. You'd be at least surprised.

> If Rofi rasi supports classes like CSS does, it could be helpful if you can have it like:

That would be easier, but it doesn't seem that `rofimoji` can attach classes to the items, unfortunately.

> I have made the [CSV for Material Symbols](https://gist.github.com/brainwo/ebebd105e81e941cd2f91ed4ca7ef561) before.

If you feel like it, please open a PR for it - especially if you have an extractor for it and not just a manually created csv.
